static/src/javascripts-legacy/projects/common/modules/commercial/membership-engagement-banner.js

```diff
@@ -160,14 +183,91 @@ define([
+            if(params.secondaryButtonCaption &&  SECONDARY_BUTTON != null && params.showRemindMe) {
+                setSecondaryButtonListener();
+            }
+
             if (messageShown) {
                 mediator.emit('membership-message:display');
             }
             mediator.emit('banner-message:complete');
         }
 
+        function emailIsValid(email){
+            return typeof email === 'string' && email.indexOf('@') > -1;
+        }
+
+        function sendEmail(email){
+            submitForm(email, LIST_ID).then(showThankYouMessage, showErrorMessage)
+        }
+
+        function showThankYouMessage(){
+            hideElement(REMIND_ME_TEXT_FIELD);
+            hideElement(REMIND_ME_CTA);
+            hideElement(REMIND_ME_ERROR);
+
+            showElement(REMIND_ME_THANKS_MESSAGE);
+        }
+
+        function showErrorMessage(){
+            hideElement(REMIND_ME_TEXT_FIELD);
+            hideElement(REMIND_ME_CTA);
+            showElement(REMIND_ME_ERROR);
+        }
+
+        function submitForm(email, listID) {
+            var formQueryString =
+                'email=' + encodeURI(email) + '&' +
+                'listId=' + listID;
+
+            return fetch(config.page.ajaxUrl + EMAIL_PATH,
+                        {   method: 'post',
+                            body: formQueryString,
+                            headers: {
+                                'Accept': 'application/json'
+                            }
+                        }
+            );
+        }
+
+        function getDOMElements(){
+            REMIND_ME_FORM = document.querySelector(REMIND_ME_FORM_SELECTOR);
+            REMIND_ME_TEXT_FIELD = document.querySelector(REMIND_ME_TEXT_FIELD_SELECTOR);
+            REMIND_ME_CTA = document.querySelector(REMIND_ME_CTA_SELECTOR);
+            REMIND_ME_THANKS_MESSAGE = document.querySelector(REMIND_ME_THANKS_MESSAGE_SELECTOR);
+            REMIND_ME_ERROR = document.querySelector(REMIND_ME_ERROR_SELECTOR);
+        }
+
+        function showElement(element) {
+            element.classList.remove('hide-element');
+        }
+
+        function hideElement(element) {
+            element.classList.add('hide-element');
+        }
+
+        function setSecondaryButtonListener() {
+            getDOMElements();
+
+            SECONDARY_BUTTON.addEventListener('click', function () {
+                hideElement(SECONDARY_BUTTON);
+                showElement(REMIND_ME_FORM);
+            })
+
+            REMIND_ME_CTA.addEventListener('click', function () {
+                var email = REMIND_ME_TEXT_FIELD.value;
+                if(emailIsValid(email)){
+                    sendEmail(email);
+                }else{
+                    showElement(REMIND_ME_ERROR);
+                }
+            });
+        }
+
         function init() {
             var bannerParams = deriveBannerParams();
+            SECONDARY_BUTTON = document.querySelector(SECONDARY_BUTTON_SELECTOR);
```

Comment:
Does `SECONDARY_BUTTON` need to be assigned a value at this point? Could this be done within one of the functions that uses it? Does it need to be scoped outside of the function that uses it?